build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -91,7 +91,7 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             // TODO: This breaks the provider
             options.links(javadocs.getJavaApi().get().toString(), javadocs.getGroovyApi().get().toString());
 
-            task.source(extension.getDocumentedSource());
+            task.source(extension.getDocumentedSource().filter(f -> f.getName().endsWith(".java") || f.getName().endsWith(".kt")));
```

Comment:
🤔 Should it also include `.groovy` because of https://github.com/gradle/gradle/blob/c670e01eebc514b8aec58371a7a1188b87146f2e/build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts#L96 ?

🤔 Why is explicit filtering needed here? It wasn't here before. What would happen without it?